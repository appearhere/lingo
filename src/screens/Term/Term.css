.top {
  max-width: 45rem;
}

.bottom {
  display: grid;
  grid-gap: calc(var(--size-lg-i) + var(--size-sm-ii));
  min-height: 0;
  min-width: 0;
  margin-top: calc(var(--size-lg-i) + var(--size-sm-ii));
}

.section {
  overflow: hidden;
  min-width: 0;
  min-height: 0;
}

.h1 {
  margin-top: 0;
  margin-bottom: 0;
  font-weight: var(--fontweight-demi);
  font-size: var(--fontsize-large-i);
}

.h2 {
  margin-top: 0;
  margin-bottom: 0;
  font-weight: var(--fontweight-demi);
  font-size: var(--fontsize-regular);
  white-space: nowrap;
}

.h2:after {
  margin-top: var(--size-regular);
  margin-bottom: var(--size-regular);
  background-color: var(--color-black);
  height: 1px;
  bottom: 0;
  content: '';
  display: block;
  left: 0;
  vertical-align: middle;
  width: 100%;
}

.h3 {
  margin-top: var(--size-regular);
  margin-bottom: var(--size-regular);
  font-size: var(--fontsize-small-i);
}

.h3:first-of-type {
  margin-top: 0;
}

.body p {
  font-weight: var(--fontweight-regular);
  font-size: var(--fontsize-small-i);
}

.badge {
  margin-top: var(--size-regular);;
}

.definition p {
  margin-top: var(--size-regular);
  font-weight: var(--fontweight-demi);
  font-size: var(--fontsize-regular);
}

@media(min-width: 36.25rem) {
  .bottom {
    grid-template-columns: repeat(2, 1fr);
    grid-gap: var(--size-lg-iii);
    margin-top: var(--size-lg-iii)
  }

  .h1 {
    font-size: var(--fontsize-large-v);
  }

  .definition p {
    font-size: var(--fontsize-large-i);
  }
}

@media(min-width: 48.75rem) {
  .bottom {
    grid-template-columns: repeat(3, 1fr);
  }
}
