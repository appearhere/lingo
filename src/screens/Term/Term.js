import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import Markdown from '@appearhere/bloom/components/Markdown/Markdown';

import terms from '../../utils/terms';

import sharedCss from '../../shared.css';
import css from './Term.css';

const Term = ({ match }) => {
  const term = terms.find(
    t => (t.name === match.params.term) && (t.department === match.params.department)
  ) || {};

  return (
    <div className={ cx(sharedCss.container, css.root) }>
      <div className={ css.top }>
        <h1 className={ css.h1 }>{ term.name }</h1>
        { term.definition && (
          <Markdown className={ css.definition }>
            { term.definition }
          </Markdown>
        ) }
      </div>
      <div className={ css.bottom }>
        <div className={ css.section }>
          <h2 className={ css.h2 }>What</h2>
          { term.what && (
            <Markdown className={ css.body }>
              { term.what }
            </Markdown>
          ) }
        </div>
        <div className={ css.section }>
          <h2 className={ css.h2 }>When</h2>
          { term.when && (
            <Markdown className={ css.body }>
              { term.when }
            </Markdown>
          ) }
        </div>
        <div className={ css.section }>
          <h2 className={ css.h2 }>Where</h2>
          { term.where && (
            <Markdown className={ css.body }>
              { term.where }
            </Markdown>
          ) }
        </div>
        <div className={ css.section }>
          <h2 className={ css.h2 }>Expected Boundary</h2>
          { term.expected_boundary && (
            <Markdown className={ css.body }>
              { term.expected_boundary }
            </Markdown>
          ) }
        </div>
        <div className={ css.section }>
          <h2 className={ css.h2 }>Hint</h2>
          { term.hint && (
            <Markdown className={ css.body }>
              { term.hint }
            </Markdown>
          ) }
        </div>
        { term.looker && (
          <div className={ css.section }>
            <h2 className={ css.h2 }>Looker</h2>
            <div className={ css.linkContainer }>
              <a className={ css.link } href={ term.looker } target="_blank" rel="noopener noreferrer">
                { term.looker }
              </a>
            </div>
          </div>
        ) }
      </div>
    </div>
  );
};

Term.propTypes = {
  match: PropTypes.shape({}).isRequired,
};

export default Term;
