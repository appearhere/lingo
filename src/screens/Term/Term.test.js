import React from 'react';
import renderer from 'react-test-renderer';
import { StaticRouter, Route } from 'react-router';

import Term from './Term';

jest.mock('../../utils/terms');

describe('Term component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<StaticRouter><Route component={ Term } /></StaticRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
