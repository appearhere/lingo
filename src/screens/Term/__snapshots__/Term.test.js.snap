// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Term component renders correctly 1`] = `
<div
  className=""
>
  <div
    className={undefined}
  >
    <h1
      className={undefined}
    />
  </div>
  <div
    className={undefined}
  >
    <div
      className={undefined}
    >
      <h2
        className={undefined}
      >
        What
      </h2>
    </div>
    <div
      className={undefined}
    >
      <h2
        className={undefined}
      >
        When
      </h2>
    </div>
    <div
      className={undefined}
    >
      <h2
        className={undefined}
      >
        Where
      </h2>
    </div>
    <div
      className={undefined}
    >
      <h2
        className={undefined}
      >
        Expected Boundary
      </h2>
    </div>
    <div
      className={undefined}
    >
      <h2
        className={undefined}
      >
        Hint
      </h2>
    </div>
  </div>
</div>
`;
